kernel/src/engine/arrow_utils.rs

```diff
@@ -677,6 +678,7 @@ fn compute_nested_null_masks(sa: StructArray, parent_nulls: Option<&NullBuffer>)
 
 /// Arrow lacks the functionality to json-parse a string column into a struct column -- even tho the
 /// JSON file reader does exactly the same thing. This function is a hack to work around that gap.
+#[internal_api]
 pub(crate) fn parse_json(
```

Comment:
do we need to make this `#[internal_api]`? doesn't the DefaultJsonHandler itself's `parse_json` just call this function? so then you could just use that?

Reply:
I wanted to avoid creating a dedicated instance of the json hander just to proxy this function. The hope was that eventually (one day) this would move upstream, so i thought it's not a huge commitment to upkeep? The json read side is fundamentally different than the default one.